### PR TITLE
Add audit log entries for developer mode, remote editor access and device snapshot

### PR DESCRIPTION
**Pull request by Steve-Mcl**
## Description

Adds audit log entries for developer mode, remote editor access and device snapshot

image

image

Adds tests for new loggers:

* Provides a logger for creating a snapshot from a project device
* Provides a logger for when developer mode is enabled for a device in a team
* Provides a logger for when developer mode is disabled for a device in a team
* Provides a logger for when remote access is enabled for a device in a team
* Provides a logger for when remote access is disabled for a device in a team

image
image

## Related Issue(s)

[#2115](https://github.com/flowforge/flowforge/issues/2115)

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 

## Labels

 - [-] Backport needed? -> add the `backport` label
 - [-] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/auditLog/team.js
Comment (joepavitt) [inline forge/auditLog/team.js:1]: being picky, but to be consistent with others, shouldn't they be:

```
developerMode.enabled
developerMode.disabled
```

and

```
remoteAccess.enabled
remoteAccess.disabled
```

Comment (Steve-Mcl) [reply forge/auditLog/team.js:1]: yes, they should
1 minute pls

Comment (Steve-Mcl) [reply forge/auditLog/team.js:1]: actually, can you point to where they are `enable` / `disable` (and not `enabled` / `disabled`)

I can't see them?

**Review by joepavitt — COMMENTED** [review]
Mentioned as a single comment - think the nested audit log approach we have means it should be:

```
developerMode.enabled
```

rather than 

```
developerModeEnabled
```

It's picky, but consistency is key and all that...

File: forge/auditLog/team.js
Comment (Steve-Mcl) [reply forge/auditLog/team.js:1]: resolved in https://github.com/flowforge/flowforge/pull/2119/commits/423af78205ea1cbb4292032395b7b1fac0206627

**Pull request merged**